### PR TITLE
[ustate] api to declare binders and purge them

### PR DESCRIPTION
**Pull request by gares**
This are the APIs which I'm currently implementing with Obj.magic to make https://github.com/LPCIC/coq-elpi/pull/315 work

### CODE REVIEW

File: engine/uState.ml [hunk 1]
```diff
@@ -574,6 +579,12 @@ let restrict uctx vars =
   let uctx' = restrict_universe_context ~lbound:uctx.universes_lbound uctx.local vars in
   { uctx with local = uctx' }
 
+let restrict_even_binders uctx vars =
+  let names, names_rev = uctx.names in
+  let names = Names.Id.Map.filter (fun _ l -> Univ.Level.Set.mem l vars) names in
+  let names_rev = Univ.Level.Map.filter (fun l _ -> Univ.Level.Set.mem l vars) names_rev in
+  restrict { uctx with names = (names,names_rev) } vars
```
Comment (SkySkimmer) [inline engine/uState.ml:586]: Why not
```suggestion
  let vars = Level.Set.union vars uctx.seff_univs in
  let uctx' = restrict_universe_context ~lbound:uctx.universes_lbound uctx.local vars in
  { uctx with local = uctx' }
```
?
ie same as `restrict` but without adding named univs to `vars`

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
Implementation looks wrong

**Comment by gares** [comment]
Thanks, your code works for me

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
squash?

**Comment by gares** [comment]
done

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**